llvm_jni/modules/jni.llvm/Type.bal

```diff
@@ -0,0 +1,161 @@
+import ballerina/jballerina.java;
```

Comment:
Some of this code looks very similar to what's in the existing nback.llvm. We need to find a way to factor things so that we avoid duplicated code.

Reply:
I think it would be best if we move string union types out of the nback.llvm in to a common place so that both jni.llvm and nback.llvm can use them without having to redefine then seperatly.

Reply:
Difficulty is that we do not want things to be less convenient for the user of the llvm module. I don't want build.bal to have to deal with two llvm modules (one with the shared types and one with everything else).